app/main/activity/crowdsourced-survey-table.directive.js

```diff
@@ -0,0 +1,97 @@
+module.exports = CrowdsourcedSurveyTable;
+
+function CrowdsourcedSurveyTable() {
+    return {
+        restrict: 'E',
+        scope: {
+            filters: '='
+        },
+        controller: CrowdsourcedSurveyTableController,
+        template: require('./crowdsourced-survey-table.html')
+    };
+}
+
+CrowdsourcedSurveyTableController.$inject = ['$scope', '$translate', 'FormEndpoint', 'FormStatsEndpoint', 'FormContactEndpoint', 'PostEndpoint', '_', 'PostFilters', 'Session'];
+function CrowdsourcedSurveyTableController($scope, $translate, FormEndpoint, FormStatsEndpoint, FormContactEndpoint, PostEndpoint, _, PostFilters, Session) {
+    $scope.crowdsourcedSurveyStatsByForm = [];
+    $scope.totalAllReduced = 0;
+    $scope.totalWebReduced = 0;
+    $scope.totalEmailReduced = 0;
+    $scope.totalSmsReduced = 0;
+    $scope.totalTwitterReduced = 0;
+    $scope.noSurveys = false;
+    $scope.unstructuredStats = {};
+
+    // whenever the filters changes, update the current list of posts
+    // this also initiates the first call to get form stats because
+    // it is called when the filters load the first time
+    $scope.$watch('filters', function () {
+        clearStats();
+        getFormStats();
+    }, true);
+    PostFilters.setMode('activity');
+
+    function clearStats() {
+        $scope.crowdsourcedSurveyStatsByForm = [];
+        $scope.totalAllReduced = 0;
+        $scope.totalWebReduced = 0;
+        $scope.totalEmailReduced = 0;
+        $scope.totalSmsReduced = 0;
+        $scope.totalTwitterReduced = 0;
+        $scope.unstructuredStats = {};
+    }
+
+    function getFormStats() {
+        FormEndpoint.query().$promise.then((forms) => {
+            if (crowdsourcedSurveysExist) {
```

Comment:
you can do if (!crowdsourcedSurveysExist) { 
                $scope.noSurveys = true;
return; 
} 
{.... code that was in the crowdsourcedSurveysExist  if .... }
To make it a little easier to read. But also feel free to ignore if you think it's not worth it / disagree

Reply:
also I think it should be crowdsourcedSurveysExist(forms) ? right?